packages/synapse-react-client/src/components/entity/metadata/EntityModal.tsx

```diff
@@ -184,18 +201,56 @@ export function EntityModal(props: EntityModalProps) {
     </>
   )
 
+  const cancelEditingConfirmationModal = (
+    <ConfirmationDialog
+      open={showEditCancellationWarning}
+      title="Unsaved Changes"
+      content="Any unsaved changes will be lost. Are you sure you want to close the editor?"
+      onCancel={() => {
+        setShowEditCancellationWarning(false)
+      }}
+      onConfirm={() => {
+        setShowEditCancellationWarning(false)
+        onCancelEdit()
+      }}
+    />
+  )
+
+  const modalTitle = useMemo(() => {
+    if (!entityBundle?.entity.name) {
+      return <Skeleton width={'40%'} />
+    } else if (currentTab === 'ANNOTATIONS') {
+      return `${isInEditMode ? 'Edit ' : ''}Annotations for ${
+        entityBundle.entity.name
+      }`
+    }
+    return entityBundle.entity.name
+  }, [currentTab, entityBundle?.entity.name, isInEditMode])
+
   return (
     <DialogBase
       className={`EntityMetadata`}
       open={show}
-      onCancel={onClose}
+      onCancel={() => {
+        if (isInEditMode) {
+          setShowEditCancellationWarning(true)
+          setOnCancelEdit(() => {
+            // Need to return a function in a function because passing a function to setState has a different meaning
+            return () => {
+              setIsInEditMode(false)
+              onClose()
+            }
+          })
```

Comment:
If this button is clicked while in edit mode, then we need to close the entire modal if the user confirms they want to discard changes